File: src/main/java/com/pygmalios/reactiveinflux/jova/wrapper/JavaReactiveInflux.java
package com.pygmalios.reactiveinflux.jova.wrapper;

import com.pygmalios.reactiveinflux.jova.PingResult;
import com.pygmalios.reactiveinflux.jova.ReactiveInflux;
import com.pygmalios.reactiveinflux.jova.ReactiveInfluxConfig;
import com.pygmalios.reactiveinflux.jova.ReactiveInfluxDb;

import java.io.IOException;
import java.util.concurrent.Future;

class JavaReactiveInflux implements ReactiveInflux {
    private final com.pygmalios.reactiveinflux.ReactiveInflux reactiveInflux;

    JavaReactiveInflux(final com.pygmalios.reactiveinflux.ReactiveInflux reactiveInflux) {
        this.reactiveInflux = reactiveInflux;
    }

    @Override
    public final Future<PingResult> ping() {
        return null;
    }

    @Override
    public final ReactiveInfluxDb database(final String dbName) {
        return null;
    }

    @Override
    public final ReactiveInfluxConfig getConfig() {
        return null;
    }

    @Override
    public void close() throws IOException {
    }
}

File: src/main/scala/com/pygmalios/reactiveinflux/jawa/wrapper/JavaReactiveInflux.scala
package com.pygmalios.reactiveinflux.jawa.wrapper

import com.pygmalios.reactiveinflux.jawa._
import com.pygmalios.{reactiveinflux => sc}
import play.libs.F.Promise

import scala.concurrent.ExecutionContext.Implicits.global

private[jawa] class JavaReactiveInflux(underlying: sc.ReactiveInflux) extends ReactiveInflux {
  override def ping(): Promise[PingResult] = Promise.wrap(underlying.ping().map(Conversions.toJava))
  override def database(dbName: String): ReactiveInfluxDb = ??? //TODO: underlying.database(new ReactiveInfluxDbName(dbName))
  override def getConfig: ReactiveInfluxConfig = ??? // TODO:
  override def close() = underlying.close()

  override def toString: String = underlying.toString
  override def hashCode(): Int = underlying.hashCode()
  override def equals(obj: scala.Any): Boolean = underlying.equals(obj)
}
